### PR TITLE
fix(android): crash when activity context not available

### LINKED ISSUE #432: java.lang.NullPointerException ReactNativeGoogleMobileAdsBannerAdViewManager.initAdView
### What happened?

app crashing in android

 io.invertase.googlemobileads.ReactNativeGoogleMobileAdsBannerAdViewManager.initAdView (ReactNativeGoogleMobileAdsBannerAdViewManager.java:174)
 
 I use   "react-native-google-mobile-ads": "^11.6.3" version
 

### Platforms

Only on Android

### React Native Info

```Shell
System:
    OS: macOS 13.4
    CPU: (8) arm64 Apple M1
    Memory: 38.42 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.2.0 - /opt/homebrew/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 9.6.6 - /opt/homebrew/bin/npm
    Watchman: 2023.05.22.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: 1.12.1 - /opt/homebrew/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 22.4, iOS 16.4, macOS 13.3, tvOS 16.4, watchOS 9.4
    Android SDK: Not Found
  IDEs:
    Android Studio: 2022.2 AI-222.4459.24.2221.10121639
    Xcode: 14.3/14E222b - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.19 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.68.4 => 0.68.4 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

### Are your using Typescript?

- [ ] My project is using Typescript

### package.json

```JSON
"react": "17.0.2",
    "react-native": "0.68.4",
    "react-native-device-info": "^10.2.0",
    "react-native-fast-image": "^8.3.7",
    "react-native-gesture-handler": "^2.10.0",
    "react-native-google-mobile-ads": "^11.6.3",
```

### app.json

```JSON
"name": "xxx",
  "displayName": "xxx",
```

### ios/Podfile

_No response_

### android/build.gradle

_No response_

### android/app/build.gradle

_No response_

### android/settings.gradle

_No response_

### AndroidManifest.xml

_No response_

### PR DESCRIPTION
**Pull request by Sygn53**
### Description
Fixes #432

### Checklist

- I read the [Contributor Guide](https://github.com/invertase/react-native-google-mobile-ads/blob/main/CONTRIBUTING.md)
  and followed the process outlined there for submitting PRs.
  - [x] Yes
- My change supports the following platforms;
  - [x] `Android`
  - [ ] `iOS`
- My change includes tests;
  - [x] `e2e` tests added or updated in `__tests__e2e__`
  - [ ] `jest` tests added or updated in `__tests__`
- [x] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [ ] Yes
  - [x] No

### Test Plan

---

Think `react-native-google-mobile-ads` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`Invertase`](https://twitter.com/invertaseio) on Twitter

### CODE REVIEW

**Review by mikehardy — COMMENTED** [review]
Interesting! This seems reasonable, the current Activity *can* go null in react-native apps due to the asynchronous nature of the react-native bridge. It is rare but it can happen.

So the code change seems valid although there is some spacing inconsistency (running `yarn lint:android`  (defined [here](https://github.com/invertase/react-native-google-mobile-ads/blob/1598969db5010afdc0850036bd10198c2d3e209f/package.json#L71)) and committing the result might fix it automatically?) and it would probably help developers to log a warning ("warning" level chosen because this is a serious problem that we cannot really handle without disabling functionality - we do not fetch an ad, for instance, but we can recover at least and not crash)

File: android/src/main/java/io/invertase/googlemobileads/ReactNativeGoogleMobileAdsBannerAdViewManager.java [hunk 1]
```diff
@@ -167,15 +167,21 @@ private BaseAdView initAdView(ReactNativeAdView reactViewGroup) {
...
-      // in order to display the debug menu for GAM ads we need the activity context
-      // https://github.com/invertase/react-native-google-mobile-ads/issues/188
-      adView =
-          new AdManagerAdView(
-              Objects.requireNonNull(
-                  ((ReactContext) reactViewGroup.getContext()).getCurrentActivity()));
-    } else {
-      adView = new AdView(reactViewGroup.getContext());
-    }
+          // in order to display the debug menu for GAM ads we need the activity context
```
Comment (dylancom) [inline android/src/main/java/io/invertase/googlemobileads/ReactNativeGoogleMobileAdsBannerAdViewManager.java:170]: Please remove the duplicated comment:
```
// in order to display the debug menu for GAM ads we need the activity context
// https://github.com/invertase/react-native-google-mobile-ads/issues/188
```

**Review by dylancom — CHANGES_REQUESTED** [review]

**Comment by dylancom** [comment]
Fixes #432

**Pull request merged**